creed/rosetta.py

```diff
@@ -0,0 +1,290 @@
+                          save_format='normalized', batch_size=10, gaus_rad=1, norm_const=100):
+    """Function to compensate MIBI data with a flow-cytometry style compensation matrix
+
+    Args:
+        raw_data_dir: path to directory containing raw images
+        comp_data_dir: path to directory where compensated images will be saved
+        comp_mat_path: path to compensation matrix, nxn with channel labels
+        panel_info_path: path to panel info file with 'masses' and 'targets' as columns
+        save_format: flag to control how the processed tifs are saved. Must be one of:
+            'raw': Direct output from the compensation matrix corresponding to number of ion events
+                detected per pixel. These will not be viewable in many image processing programs
+            'normalized': all images are divided by 100 to enable visualization. This transform
+                has no effect on downstream analysis as it preserves relative expression values
+            'both': saves both 'raw' and 'normalized' images
+        batch_size: number of images to process at a time
+        gaus_rad: radius for blurring image data. Passing 0 will result in no blurring
+        norm_const: constant used for normalization if save_format == 'normalized'
+    """
+
+    # get list of all fovs
+    fovs = list_folders(raw_data_dir, substrs=['fov'])
+
+    # load csv files
+    comp_mat = pd.read_csv(comp_mat_path, index_col=0)
+    panel_info = pd.read_csv(panel_info_path)
+    acquired_masses = panel_info['masses'].values
+    acquired_targets = panel_info['targets'].values
+    all_masses = comp_mat.columns.values.astype('int')
+
+    # make sure panel is in increasing order
+    if not np.all(acquired_masses == sorted(acquired_masses)):
+        raise ValueError("Masses must be sorted numerically in the panel_info file")
+
+    # make sure channels in comp matrix are same as those in panel csv
+    verify_same_elements(acquired_masses=acquired_masses, compensation_masses=all_masses)
+
+    # check first FOV to make sure all channels are present
+    test_data = load_imgs_from_tree(data_dir=raw_data_dir, fovs=fovs[0:1],
+                                    channels=acquired_targets, dtype='float32')
+
+    verify_same_elements(image_files=test_data.channels.values, listed_channels=acquired_targets)
+
+    # check for valid save_formats
+    allowed_formats = ['raw', 'normalized', 'both']
+    verify_in_list(save_format=save_format, allowed_formats=allowed_formats)
+
+    if batch_size < 1 or not isinstance(batch_size, int):
+        raise ValueError('batch_size parameter must be a positive integer')
+
+    if gaus_rad < 0 or not isinstance(gaus_rad, int):
+        raise ValueError('gaus_rad parameter must be a non-negative integer')
+
+    # loop over each set of FOVs in the batch
+    for i in range(0, len(fovs), batch_size):
+        print("Processing image {}".format(i))
```

Comment:
Would it be better to print the name of the FOV (ex. `print("Processing image {}".format(fovs[i]))` rather than the index?

Reply:
These are the instrument names, meaning it's just fov1, fov2, etc